src/tool/core/mod.rs

```diff
@@ -18,12 +20,20 @@ use test::do_test;
 pub fn run(cli: &Cli) -> anyhow::Result<()> {
     cli.update_current_working_dir()?;
 
-    working_directory_validation()?;
-
     match &cli.command {
-        cli::Commands::Generate(args) => do_generate(args),
-        cli::Commands::Active(args) => do_active(args),
-        cli::Commands::Test => do_test(),
+        cli::Commands::Generate(args) => {
+            working_directory_validation()?;
```

Comment:
This is necessary because no `Cargo.toml` can and should be found when doing `cargo leet new`

Reply:
Ok, it would be 3 lines less, but would hurt the readability. An `if` expression achieving the same this as the `match` after it could, would feel weird. And if we'd ever want to add, let's say an `init` subcommand, we'd have to make the `if` condition longer, and make it even less readable.

Even if there's a bit of repetition, the code is still really readable

Reply:
Ok agreed.